### ISSUE TITLE
When converting a QB query to SQL, prepared statement params aren't substituted

### ISSUE THREAD

**Issue by tlrobinson** [metabase/metabase#3549]
We don't substitute prepared statements parameters when converting to SQL, e.x.

``` sql
SELECT count(*) AS "count"
FROM "PUBLIC"."PRODUCTS"
WHERE "PUBLIC"."PRODUCTS"."CATEGORY" = ?
```

This results in a somewhat confusing `Parameter "#1" is not set` error.

Ideally we should substitute the actually parameters, but if we can't do that we should provide a more helpful warning of some kind.

**Comment by camsaul** [comment]
Yeah. I think ideally we'd put them in the correct spots in the SQL but that's not a small task since something like a `java.sql.Timestamp` would be represented differently in text on pretty much SQL DB we support.

In the meantime a warning would be a good idea however 👍

**Comment by mazameli** [comment]
Yeah, I'd love to add a warning to this. As a novice SQL writer, this really threw me for a loop this morning.

**Comment by camsaul** [comment]
Should be easier to do once #4325 is merged

**Comment by camsaul** [comment]
Implemented by #9460
